### PR TITLE
171 camera panning

### PR DESCRIPTION
**Pull request by MrJackey**
## Summary
Fixes 171
Added a cooldown on camera panning
Now if a lever activates multiple gates, they all are shown.

Built upon #64 and should be merged after it

### CODE REVIEW

File: Assets/Scripts/Puzzle/Gate.cs [hunk 1]
```diff
@@ -29,15 +35,17 @@ public class Gate : SignalReceiver {
...
+				StartCoroutine(CoShowEvent());
+		}
 		else if (animator.isInitialized)
 			animator.SetBool(isOpenHash, IsActivated);
-
-		allowShow = true;
 	}
 
-	private IEnumerator ShowEvent() {
+	public IEnumerator CoShowEvent() {
```
Comment (smeas) [inline Assets/Scripts/Puzzle/Gate.cs]: Can be private.

File: Assets/Scripts/Puzzle/Gate.cs [hunk 2]
```diff
@@ -10,15 +11,20 @@ public class Gate : SignalReceiver {
 	[Header("Gate")]
 	[SerializeField] private Animator animator;
 	[SerializeField] private bool panCamera;
-	[SerializeField] private float showDuration = 1f;
+	[SerializeField, EnableIf(nameof(panCamera))]
+	private float showDuration = 1f;
+	[SerializeField, EnableIf(nameof(panCamera))]
+	private float showCooldown = 7.5f;
+
+	private static List<Gate> panningQueue = new List<Gate>();
```
Comment (smeas) [inline Assets/Scripts/Puzzle/Gate.cs]: Using a `Queue<T>` here would be more efficient.

**Review by smeas — CHANGES_REQUESTED** [review]
It would also be nice to be able to set some kind of priority/ordering for the gates so you can control in which order they are shown. But this is probably good for now.

File: Assets/Scripts/Puzzle/Gate.cs (hunk 2 above)
Comment (MrJackey) [reply Assets/Scripts/Puzzle/Gate.cs]: Of course that's a thing

**Review by smeas — APPROVED** [review]

**Review by Addehman — APPROVED** [review]
Grande!

**Pull request merged**